tests/consensus_tests/test_hw_measurement.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import pathlib
+
+from .fixtures import create_collection, upsert_random_points, get_telemetry_hw_info, update_points_payload
+from .utils import *
+from math import ceil
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger()
+
+N_PEERS = 4
+N_SHARDS = N_PEERS
+N_REPLICAS = 1
+COLLECTION_NAME = "test_collection_hw_counting"
+
+
+def test_measuring_hw_for_updates(tmp_path: pathlib.Path):
+    peer_urls, peer_dirs, bootstrap_url = start_cluster(tmp_path, N_PEERS)
+    create_collection(peer_urls[0], collection=COLLECTION_NAME, shard_number=N_SHARDS, replication_factor=N_REPLICAS)
+    wait_collection_exists_and_active_on_all_peers(collection_name=COLLECTION_NAME, peer_api_uris=peer_urls)
+    upsert_random_points(peer_urls[0], 200, collection_name=COLLECTION_NAME)
+
+    peer_hw_infos = [get_telemetry_hw_info(x, COLLECTION_NAME) for x in peer_urls]
+
+    # Check initial insertion IO measurements are reported in telemetry
+    for i in peer_hw_infos:
+        assert i["payload_io_write"] > 0
+        # TODO: Add tests for vector writes
+
+    # Upsert ~20 vectors into each shard
+    upsert_random_points(peer_urls[0], N_PEERS * 20, collection_name=COLLECTION_NAME)
```

Comment:
I think it should be this:

```suggestion
    upsert_random_points(peer_urls[0], N_SHARDS * 20, collection_name=COLLECTION_NAME)
```